### PR TITLE
Add support for adding separate addresses to an account

### PR DESCRIPTION
**Pull request by OBorce**
- new command for adding a separate public key has address to an account to be watched by the wallet
- new transactions using those addresses as inputs or outputs will be kept by the wallet

### CODE REVIEW

**Comment by iljakuklic** [comment]
Not that I have a strong opinion about it but I find it slightly weird that custom keys are under existing accounts. I'd expect it to be a separate account that holds a collection of custom keys. But there may be a good reason to do it this way. Care to elaborate about the reasoning behind the decision?

File: test/functional/wallet_watch_address.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+"""Wallet watch separate address test
+
+Check that:
+* We can create a new wallet,
+* get an address
+* send coins to the wallet's address
+* create a new wallet
+* add the an address from wallet 1 to wallet 2 to be watched
+* send coins to wallet 1's address
+* check that walelt 2 is keeping truck of transactions using that address in inputs or outputs
```
Comment (TheQuantumPhysicist) [inline test/functional/wallet_watch_address.py:26]: walelt

File: wallet/src/account/mod.rs [hunk 2]
```diff
@@ -1504,6 +1504,26 @@ impl Account {
         Ok(())
     }
 
+    /// Add a separate address not derived from this account's key chain to be watched
+    pub fn add_separate_address(
```
Comment (TheQuantumPhysicist) [inline wallet/src/account/mod.rs:1508]: Please call this standalone_addres. We agreed on the terminology.

File: wallet/storage/src/schema.rs [hunk 3]
```diff
@@ -53,5 +54,7 @@ storage::decl_schema! {
         pub DBUnconfirmedTxCounters: Map<AccountId, u64>,
         /// Store for each account's legacy VRF public key
         pub DBVRFPublicKeys: Map<AccountId, AccountVrfKeys>,
+        /// Store for separate keys added to accounts
+        pub DBSeparateKeys: Map<AccountAddress, AccountSeparateKey>,
```
Comment (TheQuantumPhysicist) [inline wallet/storage/src/schema.rs:58]: I see the naming "separate" has infected the whole codebase... please rename them all to `standalone`. The word separate isn't really the best at describing this.

File: wallet/wallet-rpc-client/src/rpc_client/client_impl.rs [hunk 4]
```diff
@@ -199,6 +199,38 @@ impl WalletInterface for ClientWalletRpc {
             .map_err(WalletRpcError::ResponseError)
     }
 
+    async fn add_separate_address(
+        &self,
+        account_index: U31,
+        address: String,
```
Comment (TheQuantumPhysicist) [inline wallet/wallet-rpc-client/src/rpc_client/client_impl.rs:205]: Why not make this `Address<Destination>`?

File: wallet/wallet-cli-lib/src/commands/mod.rs [hunk 5]
```diff
@@ -251,6 +251,24 @@ pub enum WalletCommand {
...
+    #[clap(name = "account-add-separate-address")]
+    AddSeparateKey {
+        /// The new separate address to be added to the selected account
+        address: String,
+
+        /// Optionally specify a label to the new address
+        label: Option<String>,
+    },
+
+    #[clap(name = "account-add-separate-private-key")]
```
Comment (TheQuantumPhysicist) [inline wallet/wallet-cli-lib/src/commands/mod.rs:263]: Please add `-as-hex`

File: wallet/storage/src/schema.rs
Comment (TheQuantumPhysicist) [inline wallet/storage/src/schema.rs:1]: Don't forget to bump the db version.

**Review by TheQuantumPhysicist — APPROVED** [review]
Besides the comments, seems fine. Please rename every "separate" to "standalone" and see the comments.

**Pull request closed without merging**